validator/client/runner.go

```diff
@@ -305,6 +306,35 @@ func runHealthCheckRoutine(ctx context.Context, v iface.Validator, eventsChan ch
 				return
 			}
 			isHealthy := tracker.CheckHealth(ctx)
+			if len(hosts) > 1 && features.Get().EnableBeaconRESTApi {
+				for !isHealthy {
+					cSlot, err := v.CanonicalHeadSlot(ctx)
+					if err != nil {
+						log.WithError(err).Error("Could not get canonical head slot")
+					}
+					nSlot := <-v.NextSlot()
+					for {
+						if cSlot == nSlot {
+							for i, url := range hosts {
+								if url == v.RetrieveHost() {
+									next := (i + 1) % len(hosts)
+									log.Infof("Beacon node at %s is not responding, switching to %s", url, hosts[next])
+									v.UpdateHost(hosts[next])
+									v.ProposerSettings()
```

Comment:
Was the intention here to push proposer settings to the new node? This will only return the current settings and discard the result.

Reply:
Yes, that was the intention. Not too familiar with it. I guess I will have to use `PushProposerSettings` instead which will make a bit of a mess...